### ISSUE TITLE
Quest N0B00Y16 Hand-in Regression

### ISSUE THREAD

**Issue by Interkarma** [Interkarma/daggerfall-unity#1369]
**Describe the bug**
Quest N0B00Y16 results in failure even after successfully opening box and returning to quest-giver. Quest-giver is then locked out as quest unable to terminate. Calling in @JayH2971 for some help with this one. :)

**To Reproduce**
Load below save and attempt to hand in quest.

[SAVE671.zip](https://github.com/Interkarma/daggerfall-unity/files/3445351/SAVE671.zip)

**Expected behavior**
Quest should hand in normally for reward if successful.

**Comment by Interkarma** [comment]
Apologies for tagging Ferital & Nystul earlier. I thought this was an NPC talk issue until I realised it was actually the quest itself.

**Comment by JayH2971** [comment]
Acknowledged.

**Comment by Interkarma** [comment]
In his PR to workaround this, Jay mentions the below doesn't fire in this quest.
```
_S.07_ task:
	have _crn_ set _S.08_
```
Leaving this issue open to remind to look at why this is.

**Comment by JorisVanEijden** [comment]
While running this in the debugger the HaveItem action in S.07 is checking each Update if the small chest is in the players inventory (it is) and it triggers S.08. Which does nothing because it's isComplete is already true.
So that part is working.

I could also hand in the chest at the merchant, but in one attempt I got message 1050 and 1040 at the same time and an Imp spawned. In another instance I got message 1050 and 1051 at the same time.
So something is not quite what it should be.
I'd like to continue debugging this tomorrow to learn more about the quest system.
